File: 백준/Bronze/2884. 알람 시계/알람 시계.py
a,b = list(map(int,input().split()))

if(b-45<0):
    if(a<1):
        a=23
        b= 60-45+b
    else:
        a-=1
        b = 60 - 45 + b
else:
    b-=45
print(a,b)

File: 백준/Bronze/2884. 알람 시계/알람 시계.java
import java.io.BufferedReader;
import java.io.IOException;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new java.io.InputStreamReader(System.in));
        String[] time = br.readLine().split(" ");
        int hour = Integer.parseInt(time[0]);
        int minute = Integer.parseInt(time[1]);

        if(minute < 45){
            if(hour == 0){
                hour = 23;
            }else{
                hour--;
            }
            minute += 15;
        }
        else{
            minute -= 45;
        }

        System.out.println(hour + " " + minute);
    }
}
